### DISCUSSION TITLE
Performance Issue: Nested Data Provider in Repeater Block

### DISCUSSION THREAD

**Discussion by Doctor-N0** [Budibase/budibase#11122] [Help]
I aim to display specific items of a table in fields. As shown in the figure below, I use a data provider for each field, which filters the table by e.g. the ID, value, position. Then I display the items 1-n, and can modify them in the field. For creating the data providers, I use a repeater block.

So, I only use the Data Providers to filter the Table and then display and process the filtered data. My question would be if a high number of data providers (about 20) is a performance issue, and if so, if there are alternative suggestions to perform such a filtering.

Thanks a lot!

<img src="https://github.com/Budibase/budibase/assets/105864309/6664a013-e5d0-47b0-a495-57cff5469c48" width="50%">

**Comment by melohagan** [comment]
Hey @Doctor-N0 

Why do you need to repeat the Data providers in this case? 

In any case, using a custom query to do all the necessary joins, etc to get all the data you need to display up front would be more performant.
